src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/PersistenceServiceTest.java

```diff
@@ -573,6 +598,46 @@ void shouldReturnEmptyListOfRoleAssignmentRecords()  {
         assertEquals(roleAssignmentList.size(),Integer.valueOf(0));
     }
 
+    @Test
+    void shouldReturnEmptyListOfRoleAssignmentRecordsWithRecordsPresentExistingFlagFalse() throws IOException {
+        UUID id = UUID.randomUUID();
+        AssignmentRequest assignmentRequest = TestDataBuilder
+            .buildAssignmentRequest(Status.CREATED, Status.LIVE, false);
+        RoleAssignmentEntity roleAssignmentEntity = TestDataBuilder.convertRoleAssignmentToEntity(
+            assignmentRequest.getRequestedRoles().iterator().next());
+        Optional<RoleAssignmentEntity> roleAssignmentOptional = Optional.of(roleAssignmentEntity);
+
+        when(roleAssignmentRepository.findById(id)).thenReturn(roleAssignmentOptional);
+        List<RoleAssignment> roleAssignmentList = sut.getAssignmentById(id);
+        verify(persistenceUtil, times(1))
+            .convertEntityToRoleAssignment(roleAssignmentEntity);
+        assertNotNull(roleAssignmentList);
+        assertNotNull(roleAssignmentOptional);
+        assertFalse(roleAssignmentList.isEmpty());
+        assertEquals(roleAssignmentList.size(),Integer.valueOf(1));
+
+    }
+
+    @Test
+    void shouldReturnEmptyListOfRoleAssignmentRecordsWithRecordsPresentExistingFlagTrue() throws IOException {
```

Comment:
Please rename method name.

Reply:
Changed to shouldGetEmptyListForExistingFlagTrue.